hbase-server/src/main/java/org/apache/hadoop/hbase/tool/CanaryTool.java

```diff
@@ -510,38 +510,38 @@ public Void call() {
 
     private Void readColumnFamily(Table table, ColumnFamilyDescriptor column) {
       byte[] startKey = null;
-      Get get = null;
-      Scan scan = null;
+      Scan scan = new Scan();
       ResultScanner rs = null;
       StopWatch stopWatch = new StopWatch();
       startKey = region.getStartKey();
       // Can't do a get on empty start row so do a Scan of first element if any instead.
       if (startKey.length > 0) {
```

Comment:
is there any benefit to retaining the `Get` path? I legitimately don't know.

If there is, it seems like it could be possible to make a change where we use `Get` only `if (startKey.length > 0 && !rawScanEnabled)`

But if `Get` is unnecessary, then your proposal is good, because it reduces the complexity anyway.

Reply:
I couldn't see any benefit for `Get` path.